File: Backend/src/main/java/com/backend/feature_flag_platform/Entity/Feature.java
package com.backend.feature_flag_platform.Entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;


@Entity
@Table(name = "feature")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Feature {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private  Long id ;

    private String key ;
    private  String description ;
    private  Boolean enabled = false ;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "domain_id", nullable = false)
    private Domain domain;

    @OneToMany(
            mappedBy = "feature",
            cascade = CascadeType.ALL,
            orphanRemoval = true
    )
    private List<Strategy> strategies = new ArrayList<>();

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    @PrePersist
    void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = createdAt;
    }

    @PreUpdate
    void onUpdate() {
        updatedAt = LocalDateTime.now();
    }


}

File: FrontEnd/src/modules/features/types/feature.ts
export interface Feature {
  id: number
  key: string
  description: string
  enabled: boolean
  domainId: string
  createdAt: string
  updatedAt: string
}

export interface CreateFeatureRequest {
  key: string
  description?: string
  id_domain: string
}

export interface UpdateFeatureRequest {
  key?: string
  description?: string
  id_domain?: string
}

export interface FeatureFilters extends Record<string, unknown> {
  search?: string
  page?: number
  size?: number
}
